src/vips/Image_methods.lua

```diff
@@ -175,6 +175,10 @@ function Image.new_from_memory(data, size, width, height, bands, format)
     return image
 end
 
+function Image.new_from_memory(data, width, height, bands, format)
+    return Image.new_from_memory_size(data, ffi.sizeof(data), width, height, bands, format)
```

Comment:
The `image._data = data` in `new_from_memory_size` will need to be moved here, I think, since that assumes `data` is a ffi memory area.

Reply:
I think it depends on the expected behavior from the API, because the same could apply to a string or a cdata with a GC callback to free the memory. So `new_from_memory_size` would expect a raw pointer that stay valid for the lifespan of the processing, maybe `new_from_memory_ptr` would be a better name to make the distinction clear: one is LuaJIT FFI allocated data and the other a custom pointer. 

The fact that the FFI data is referenced in the first place is confusing since the data may be used outside of the lifespan of the image, but I suppose you want backward compatibility for this behavior.